passage/src/main/java/id/passage/android/PassageOneTimePasscode.kt

```diff
@@ -0,0 +1,99 @@
+import id.passage.android.model.LoginOneTimePasscodeRequest
+import id.passage.android.model.RegisterOneTimePasscodeRequest
+import okhttp3.OkHttpClient
+
+class PassageOneTimePasscode(
+    private val passageClient: OkHttpClient,
+    private val tokenStore: PassageTokenStore,
+) {
+    /**
+     * Register with One Time Passcode
+     *
+     * Create a user and send a registration email or SMS to the user. The user will receive an
+     * email or text with a one time passcode to complete their registration.
+     * @param identifier valid email or E164 phone number
+     * @return OneTimePasscode
+     * @throws OneTimePasscodeRegisterException
+     */
+    suspend fun register(identifier: String): OneTimePasscode {
+        val registerAPI = RegisterAPI(Passage.BASE_PATH, passageClient)
+        val request = RegisterOneTimePasscodeRequest(identifier, Passage.language)
+        val response =
+            try {
+                registerAPI.registerOneTimePasscode(Passage.appId, request)
+            } catch (e: Exception) {
+                throw OneTimePasscodeRegisterException.convert(e)
+            }
+        return response
+    }
+
+    /**
+     * Login with One Time Passcode
+     *
+     * Send a login email or SMS to the user. The user will receive an email or text with a one time
+     * passcode to complete their login.
+     * @param identifier valid email or E164 phone number
+     * @return OneTimePasscode
+     * @throws OneTimePasscodeLoginException
+     */
+    suspend fun login(identifier: String): OneTimePasscode {
+        val loginAPI = LoginAPI(Passage.BASE_PATH, passageClient)
+        val request = LoginOneTimePasscodeRequest(identifier, Passage.language)
+        val response =
+            try {
+                loginAPI.loginOneTimePasscode(Passage.appId, request)
+            } catch (e: Exception) {
+                throw OneTimePasscodeLoginException.convert(e)
+            }
+        return response
+    }
+
+    /**
+     * Authenticate One Time Passcode
+     *
+     * Authenticate a one time passcode for a user. This endpoint checks that the one time passcode
+     * is valid, then returns an authentication token for the user.
+     * @param otp The one time passcode
+     * @param otpId The OTP id
+     * @return PassageAuthResult
+     * @throws OneTimePasscodeActivateException
+     */
+    suspend fun activate(
+        otp: String,
+        otpId: String,
+    ): AuthResult {
+        val otpAPI = OTPAPI(Passage.BASE_PATH, passageClient)
+        val request = ActivateOneTimePasscodeRequest(otp, otpId)
+        val response =
+            try {
+                otpAPI.activateOneTimePasscode(Passage.appId, request)
+            } catch (e: Exception) {
+                throw OneTimePasscodeActivateException.convert(e)
+            }
+        val otpAuthResult = response.authResult
+        // NOTE: The OpenAPI codegen produces an `IdentityAuthResult` for passkey and magic link
```

Comment:
This might also be fixed?

Reply:
Yes, it's fixed. thanks for pointing this out